src/geometric_algebra/generic/defs.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2020 lean-ga Team. All rights reserved.
+Released under MIT license as described in the file LICENRE.
+Authors: Utensil Rong
+-/
+import ring_theory.algebra
+import linear_algebra.quadratic_form
+import tactic
+import tactic.lint
+
+-- import geometric_algebra.generic.operators
+
+universes ur uv ug
+
+variables {R : Type ur} [field R]
+variables {V : Type uv} [add_comm_group V] [vector_space R V]
+variables {G : Type ug} [ring G]
+
+section prio
+-- set_option default_priority 200 -- see Note [default priority]
+set_option default_priority 100
+set_option old_structure_cmd true
+
+/--
+-/
+@[ancestor algebra]
+class semi_geometric_algebra
+(R : Type ur) [field R]
+(V : Type uv) [add_comm_group V] [vector_space R V]
+(G : Type ug) [ring G]
```

Comment:
It seems that I can't leave them to variables because that would cause failure to use either `semi_geometric_algebra G` or `semi_geometric_algebra R V G`. They still stick here.